src/formattedcode/output_html.py

```diff
@@ -223,6 +227,11 @@ def generate_output(results, license_references, version, template):
                     'value': license_expression,
                 })
 
+                if license_expression not in licenses:
```

Comment:
This should be done only if ``--license-references`` is not enabled which is when we don't have the license data,
```suggestion
                if not license_references and license_expression not in licenses:
```

Reply:
I added the requested conditional.